packages/integration-sdk-cli/src/commands/options.ts

```diff
@@ -0,0 +1,69 @@
+import { getApiBaseUrl } from '@jupiterone/integration-sdk-runtime';
+
+/**
+ * Determines the JupiterOne `apiBaseUrl` based on these mutually-exclusive `options`:
+ * - --api-base-url - a specific URL to use
+ * - --development - use the development API
+ *
+ * @throws Error if both --api-base-url and --development are specified
+ */
+export function getApiBaseUrlOption(options: any): string {
+  let apiBaseUrl: string;
+  if (options.apiBaseUrl) {
+    if (options.development) {
+      throw new Error(
+        'Invalid configuration supplied.  Cannot specify both --api-base-url and --development(-d) flags.',
+      );
+    }
+    apiBaseUrl = options.apiBaseUrl;
+  } else {
+    apiBaseUrl = getApiBaseUrl({
+      dev: options.development,
+    });
+  }
+  return apiBaseUrl;
+}
+
+interface SynchronizationJobSourceOptions {
+  source: 'integration-managed' | 'integration-external' | 'api';
+  scope?: string;
+  integrationInstanceId?: string;
+}
+
+/**
+ * Determines the `source` configuration for the synchronization job based on these `options`:
+ * - --source - a specific source to use
+ * - --scope - a specific scope to use, required when --source is "api"
+ * - --integrationInstanceId - a specific integrationInstanceId to use, required when --source is not "api"
```

Comment:
I should have added some `@throws` docs in here, will do as part of the versioning PR.